wordpress-mobile/WordPress-Android
# Issue/14128 related posts actions refactoring

Task: #14128

This PR is a sub-task of Issue #14128 and refactors below reader related posts actions making them testable:

1. Related post click action from post details screen.
2. Related post click action from related post details screen (replaces previous related post keeping it in history).
3. Global related post follow button action by reusing `ReaderPostCardActionsHandler`.

### To test

Test.1 Related post click action from post details screen

1. Open a wp.com Reader post from the Reader tab.
2. Click a related post on the selected post details screen. 
3. Notice that "Related Post" screen is displayed with selected related post details and with cross button.

Test.2 Related post click action from "Related Post" screen

1. Continue from above and click a related post from "Related Post" screen.
2. Notice that selected related post details are shown on "Related Post" screen.
3. Click back button.
4. Notice that previous related post (which was added to history) is displayed (as before refactoring)  on "Related Post" screen.

Test.3 Global related post follow button action (see notes)

1. Open a wp.com Reader post from the Reader tab.
2. Scroll to the global related post (from wp.com) at the bottom of the selected post details screen. 
3. Click follow button on global related post.
4. Notice that follow state is toggled for the post.

Tests included in `ReaderPostDetailViewModelTest`.

### Notes

1. Ignore all styling.
2. Test.3 can be skipped for now as we're deciding on whether to show follow button for global related posts due to limited space.

PR submission checklist:
